In [1]:
import pandas as pd

In [3]:
df = pd.read_csv("loan_data.csv")

In [4]:
df

credit.policy             purpose  int.rate  installment  \
0                 1  debt_consolidation    0.1189       829.10   
1                 1         credit_card    0.1071       228.22   
2                 1  debt_consolidation    0.1357       366.86   
3                 1  debt_consolidation    0.1008       162.34   
4                 1         credit_card    0.1426       102.92   
...             ...                 ...       ...          ...   
9573              0           all_other    0.1461       344.76   
9574              0           all_other    0.1253       257.70   
9575              0  debt_consolidation    0.1071        97.81   
9576              0    home_improvement    0.1600       351.58   
9577              0  debt_consolidation    0.1392       853.43   

      log.annual.inc    dti  fico  days.with.cr.line  revol.bal  revol.util  \
0          11.350407  19.48   737        5639.958333      28854        52.1   
1          11.082143  14.29   707        2760.000000      33623        76.7   
2          10.373491  11.63   682        4710.000000       3511        25.6   
3          11.350407   8.10   712        2699.958333      33667        73.2   
4          11.299732  14.97   667        4066.000000       4740        39.5   
...              ...    ...   ...                ...        ...         ...   
9573       12.180755  10.39   672       10474.000000     215372        82.1   
9574       11.141862   0.21   722        4380.000000        184         1.1   
9575       10.596635  13.09   687        3450.041667      10036        82.9   
9576       10.819778  19.18   692        1800.000000          0         3.2   
9577       11.264464  16.28   732        4740.000000      37879        57.0   

      inq.last.6mths  delinq.2yrs  pub.rec  not.fully.paid  
0                  0            0        0               0  
1                  0            0        0               0  
2                  1            0        0               0  
3                  1            0        0               0  
4                  0            1        0               0  
...              ...          ...      ...             ...  
9573               2            0        0               1  
9574               5            0        0               1  
9575               8            0        0               1  
9576               5            0        0               1  
9577               6            0        0               1  

[9578 rows x 14 columns]

# first we should know data information

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9578 entries, 0 to 9577
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   credit.policy      9578 non-null   int64  
 1   purpose            9578 non-null   object 
 2   int.rate           9578 non-null   float64
 3   installment        9578 non-null   float64
 4   log.annual.inc     9578 non-null   float64
 5   dti                9578 non-null   float64
 6   fico               9578 non-null   int64  
 7   days.with.cr.line  9578 non-null   float64
 8   revol.bal          9578 non-null   int64  
 9   revol.util         9578 non-null   float64
 10  inq.last.6mths     9578 non-null   int64  
 11  delinq.2yrs        9578 non-null   int64  
 12  pub.rec            9578 non-null   int64  
 13  not.fully.paid     9578 non-null   int64  
dtypes: float64(6), int64(7), object(1)
memory usage: 1.0+ MB


# -----------------------------------------------------------------------

# top 5 purpose from biggest to smallest

In [11]:
df["purpose"].value_counts().head(7)

debt_consolidation    3957
all_other             2331
credit_card           1262
home_improvement       629
small_business         619
major_purchase         437
educational            343
Name: purpose, dtype: int64

## the biggest number in purpose in debt_consolidation  there is 3957 and the smallest number in educational there is 343

-----------------------------------------------------------------------------------------

* credit.policy count

In [12]:
df["credit.policy"].value_counts().head(2)

1    7710
0    1868
Name: credit.policy, dtype: int64

there is 7710 of 1 credit.policy and 1868 of 0 credit.policy

------------------------------------------------------------------------------

* table between inq.last.6mths and delinq.2yrs and pub.rec and not.fully.paid


In [14]:
df[["inq.last.6mths","delinq.2yrs","pub.rec","not.fully.paid"]]

inq.last.6mths  delinq.2yrs  pub.rec  not.fully.paid
0                  0            0        0               0
1                  0            0        0               0
2                  1            0        0               0
3                  1            0        0               0
4                  0            1        0               0
...              ...          ...      ...             ...
9573               2            0        0               1
9574               5            0        0               1
9575               8            0        0               1
9576               5            0        0               1
9577               6            0        0               1

[9578 rows x 4 columns]

-----------------------------------------------------------------------------------

In [19]:
df["revol.bal"].max()

1207359

In [17]:
df["revol.util"].max()

119.0

------------------------------------------------------------------------

* biggest fico in which credit policy 

In [21]:
df.groupby("credit.policy").max()["fico"]

credit.policy
0    807
1    827
Name: fico, dtype: int64

in 1 has 827 